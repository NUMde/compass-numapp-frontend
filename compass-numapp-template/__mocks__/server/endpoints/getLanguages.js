import { rest } from 'msw';
import endpoints from '~services/rest/endpoints';

export default rest.get(endpoints.getLanguages, (_req, res, ctx) => {
  return res(ctx.json(['en']));
});
