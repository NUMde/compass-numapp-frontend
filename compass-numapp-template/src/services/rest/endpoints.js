import { appConfig } from '~config';

export default {
  /** rest:
   * endpoint to log the user in and retrieve the accessToken */
  login: `${appConfig.baseURI}participant/`,

  /** rest:
   * endpoint to get the user-profile */
  getUser: `${appConfig.baseURI}participant/`,

  /** rest:
   * endpoint for a special report */
  report: `${appConfig.baseURI}queue`,

  /** rest:
   * endpoint to post the questionnaire to */
  sendQuestionnaire: `${appConfig.baseURI}queue`,

  /** rest:
   * endpoint to receive the questionnaire */
  getQuestionnaire: `${appConfig.baseURI}questionnaire/`,

  /** rest:
   * endpoint to update the push token */
  updateToken: `${appConfig.baseURI}participant/update-device-token/`,

  /** rest:
   * endpoint to update the chosen language*/
  updateLanguage: `${appConfig.baseURI}participant/update-language-code/`,

  /** rest:
   * endpoint to get the list languages*/
  getLanguages: `${appConfig.baseURI}/questionnaire/get-languages/`,
};
