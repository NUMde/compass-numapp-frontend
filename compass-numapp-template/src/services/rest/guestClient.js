// (C) Copyright IBM Deutschland GmbH 2021.  All rights reserved.

// this file provides methods concerning the communication with the backend before the user
// is logged in.

/***********************************************************************************************
imports
***********************************************************************************************/

import axios from 'axios';
import { appConfig } from '../../config';

/***********************************************************************************************
client
***********************************************************************************************/

/**
 * @param  {string} subjectId the id used to identify the user
 */
const login = async (subjectId) =>
  axios.get(appConfig.endpoints.login + subjectId);

/***********************************************************************************************
export
***********************************************************************************************/

export default { login };
