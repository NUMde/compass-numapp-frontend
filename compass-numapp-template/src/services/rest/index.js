export { loggedInClient, guestClient } from './rest';
