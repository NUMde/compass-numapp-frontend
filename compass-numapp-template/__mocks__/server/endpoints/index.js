import login from './login';
import getLanguages from './getLanguages';
import updateLanguage from './updateLanguage';
import updateToken from './updateToken';
import getQuestionnaire from './getQuestionnaire';
import sendQuestionnaire from './sendQuestionnaire';

export default [
  login,
  getLanguages,
  updateLanguage,
  updateToken,
  getQuestionnaire,
  sendQuestionnaire,
];
