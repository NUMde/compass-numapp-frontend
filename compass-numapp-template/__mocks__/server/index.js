import { setupServer } from 'msw/node';
import endpoints from './endpoints';

export default setupServer(...endpoints);
