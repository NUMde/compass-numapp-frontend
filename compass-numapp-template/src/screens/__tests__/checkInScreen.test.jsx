import React from 'react';
import { Alert } from 'react-native';
import { rest } from 'msw';

import {
  renderWithRedux,
  fireEvent,
  waitFor,
  waitForElementToBeRemoved,
} from '__test-utils__/render';

import categories from '__mocks__/questionnaire/categories';
import itemMap from '__mocks__/questionnaire/itemMap';
import server from '__mocks__/server';
import { loggedInClient } from '~services/rest';

import en from '~CUSTOMIZATION/translations/en';
import { appConfig } from '~config';
import endpoints from '~services/rest/endpoints';

import CheckInScreen from '../checkInScreen';

describe('CheckInScreen', () => {
  beforeAll(() => {
    server.listen();
  });

  afterEach(() => {
    server.resetHandlers();
    jest.clearAllMocks();
  });

  afterAll(() => {
    server.close();
  });

  it('should render the CheckInScreen', (done) => {
    const navigate = jest.fn();
    const goBack = jest.fn();

    const { getByText, toJSON } = renderWithRedux(
      <CheckInScreen navigation={{ navigate, goBack }} />,
      {
        initialState: {
          User: { subjectId: 'someUserId', additional_iterations_left: 0 },
        },
      },
    );
    waitFor(() =>
      expect(getByText(en.survey.welcomeTitleFirstTime)).toBeTruthy(),
    ).then(() => {
      expect(toJSON()).toMatchSnapshot();
      done();
    });
  });

  it('should trigger user update when mounting CheckInScreen', (done) => {
    const navigate = jest.fn();
    const goBack = jest.fn();
    const restSpy = jest.spyOn(loggedInClient, 'getUserUpdate');
    const { getByText } = renderWithRedux(
      <CheckInScreen navigation={{ navigate, goBack }} />,
      {
        initialState: {
          User: { subjectId: 'someUserId', additional_iterations_left: 1 },
        },
      },
    );
    waitFor(() =>
      expect(getByText(en.survey.welcomeTitleFirstTime)).toBeTruthy(),
    ).then(() => {
      expect(restSpy).toHaveBeenCalled();
      done();
    });
  });

  it('should alert User when questionnaire is outdated', async () => {
    const navigate = jest.fn();
    const goBack = jest.fn();
    const spyAlert = jest
      .spyOn(Alert, 'alert')
      .mockImplementation(async (_title, _message, callbackOrButtons) => {
        if (callbackOrButtons && callbackOrButtons[0]) {
          callbackOrButtons[0].onPress();
        }
      });
    const { findByText, queryByText } = renderWithRedux(
      <CheckInScreen navigation={{ navigate, goBack }} />,
      {
        initialState: {
          User: {
            subjectId: 'someUserId',
            current_questionnaire_id: 'http://example.com:1.0',
            additional_iterations_left: 1,
          },
          Questionnaire: {
            itemMap,
            categories,
            FHIRmetadata: { url: 'http://exmaple.com', version: '2.0' },
          },
        },
      },
    );
    expect(await findByText(en.survey.welcomeTitleFirstTime)).toBeTruthy();
    expect(spyAlert).toHaveBeenCalledWith(
      en.generic.info,
      en.generic.infoRemoval,
      expect.anything(),
      expect.anything(),
    );
    expect(queryByText(en.survey.send)).toBeFalsy();
  });

  it('should submit completed questionnaire', (done) => {
    const navigate = jest.fn();
    const goBack = jest.fn();
    jest
      .spyOn(Alert, 'alert')
      .mockImplementation((_title, _message, callbackOrButtons) => {
        if (callbackOrButtons && callbackOrButtons[0]) {
          callbackOrButtons[0].onPress();
        }
      });
    const { findByText, getByText, queryByText, getByTestId } = renderWithRedux(
      <CheckInScreen navigation={{ navigate, goBack }} />,
      {
        initialState: {
          User: {
            subjectId: 'someUserId',
            current_questionnaire_id:
              'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel|1.0',
            status: 'on-study',
            start_date: new Date(),
            firstTime: true,
            due_date: new Date(new Date().setDate(new Date().getDate() + 3)),
            certificate: appConfig.defaultRecipientCertificatePemString,
            additional_iterations_left: 1,
          },
          Questionnaire: {
            itemMap,
            categories,
            FHIRmetadata: {
              url: 'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel',
              version: '1.0',
            },
          },
        },
      },
    );
    findByText(en.survey.send).then((btn) => {
      fireEvent.press(btn);
      waitForElementToBeRemoved(() => getByTestId('Spinner')).then(() => {
        expect(queryByText(en.survey.send)).toBe(null);
        expect(
          getByText(en.survey.noNewQuestionnaireAvailableYetTitle),
        ).toBeTruthy();
        done();
      });
    });
  });

  it('should refresh user data', (done) => {
    const navigate = jest.fn();
    const goBack = jest.fn();
    const { queryByTestId, getByTestId, findByLabelText } = renderWithRedux(
      <CheckInScreen navigation={{ navigate, goBack }} />,
      {
        initialState: {
          User: {
            subjectId: 'outdatedUser',
            current_questionnaire_id:
              'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel|1.0',
            status: 'on-study',
            start_date: new Date(),
            firstTime: true,
            due_date: new Date(new Date().setDate(new Date().getDate() + 3)),
            certificate: appConfig.defaultRecipientCertificatePemString,
            additional_iterations_left: 0,
          },
          Questionnaire: {
            itemMap,
            categories,
            FHIRmetadata: {
              url: 'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel',
              version: '1.0',
            },
          },
        },
      },
    );
    findByLabelText(en.accessibility.refresh).then((button) => {
      expect(getByTestId('CheckInListItem')).toBeTruthy();
      expect(button).toBeTruthy();
      fireEvent.press(button);
      waitForElementToBeRemoved(() => getByTestId('Spinner')).then(() => {
        expect(queryByTestId('CheckInListItem')).toBe(null);
        done();
      });
    });
  });

  it('should notify that a report has already been sent out', async () => {
    const spyAlert = jest
      .spyOn(Alert, 'alert')
      .mockImplementation(async (_title, _message, callbackOrButtons) => {
        if (callbackOrButtons && callbackOrButtons[0]) {
          callbackOrButtons[0].onPress();
        }
      });
    const navigate = jest.fn();
    const goBack = jest.fn();
    const { findByText } = renderWithRedux(
      <CheckInScreen navigation={{ navigate, goBack }} />,
      {
        initialState: {
          User: {
            subjectId: 'someUserId',
            current_questionnaire_id:
              'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel|1.0',
            status: 'on-study',
            start_date: new Date(),
            firstTime: true,
            due_date: new Date(new Date().setDate(new Date().getDate() + 3)),
            certificate: appConfig.defaultRecipientCertificatePemString,
            additional_iterations_left: 1,
          },
          Questionnaire: {
            itemMap,
            categories,
            FHIRmetadata: {
              url: 'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel',
              version: '1.0',
            },
          },
        },
      },
    );
    const button = await findByText(en.reporting.symptoms_header);
    expect(button).toBeTruthy();
    fireEvent.press(button);
    expect(spyAlert).toHaveBeenCalledWith(
      en.generic.info,
      en.generic.reportWhileInIteratedMode,
      expect.anything(),
      expect.anything(),
    );
  });

  it('should notify that a questionnaire is available to report symptoms', async () => {
    const spyAlert = jest
      .spyOn(Alert, 'alert')
      .mockImplementation(async (_title, _message, callbackOrButtons) => {
        if (callbackOrButtons && callbackOrButtons[0]) {
          callbackOrButtons[0].onPress();
        }
      });
    const navigate = jest.fn();
    const goBack = jest.fn();
    const { findByText } = renderWithRedux(
      <CheckInScreen navigation={{ navigate, goBack }} />,
      {
        initialState: {
          User: {
            subjectId: 'someUserId',
            current_questionnaire_id:
              'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel|1.0',
            status: 'on-study',
            start_date: new Date(),
            firstTime: true,
            due_date: new Date(new Date().setDate(new Date().getDate() + 3)),
            certificate: appConfig.defaultRecipientCertificatePemString,
            additional_iterations_left: 0,
          },
          Questionnaire: {
            itemMap,
            categories,
            FHIRmetadata: {
              url: 'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel',
              version: '1.0',
            },
          },
        },
      },
    );
    const button = await findByText(en.reporting.symptoms_header);
    expect(button).toBeTruthy();
    fireEvent.press(button);
    expect(spyAlert).toHaveBeenCalledWith(
      en.generic.info,
      en.generic.reportWhileQuestionnaire,
      expect.anything(),
      expect.anything(),
    );
  });

  it('should send report', (done) => {
    const spyAlert = jest
      .spyOn(Alert, 'alert')
      .mockImplementation(async (_title, _message, callbackOrButtons) => {
        if (callbackOrButtons && callbackOrButtons[0]) {
          callbackOrButtons[0].onPress();
        }
      });
    const navigate = jest.fn();
    const goBack = jest.fn();
    const { findByText, getByTestId, queryByTestId, getByText } =
      renderWithRedux(<CheckInScreen navigation={{ navigate, goBack }} />, {
        initialState: {
          User: {
            subjectId: 'reportingUser',
            current_questionnaire_id:
              'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel|1.0',
            status: 'on-study',
            start_date: new Date(new Date().setDate(new Date().getDate() + 2)),
            firstTime: true,
            due_date: new Date(new Date().setDate(new Date().getDate() + 3)),
            certificate: appConfig.defaultRecipientCertificatePemString,
            additional_iterations_left: 0,
          },
          Questionnaire: {
            itemMap,
            categories,
            FHIRmetadata: {
              url: 'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel',
              version: '1.0',
            },
          },
        },
      });
    findByText(en.reporting.symptoms_header).then((button) => {
      expect(queryByTestId('CheckInListItem')).toBe(null);
      expect(button).toBeTruthy();
      fireEvent.press(button);
      expect(spyAlert).toHaveBeenCalledWith(
        en.reporting.symptoms_header,
        en.reporting.symptoms_question,
        expect.anything(),
        expect.anything(),
      );
      waitForElementToBeRemoved(() => getByTestId('Spinner')).then(() => {
        expect(getByTestId('CheckInListItem')).toBeTruthy();
        expect(getByText(en.survey.surveyTitle)).toBeTruthy();
        done();
      });
    });
  });

  it('should alert when user update failed', (done) => {
    const spyAlert = jest.spyOn(Alert, 'alert');
    const navigate = jest.fn();
    const goBack = jest.fn();
    server.use(
      rest.get(`${endpoints.login}:subjectId`, (_req, res, ctx) => {
        return res(
          ctx.status(404),
          ctx.json({
            errorCode: 'InternalErr',
            errorMessage: 'An internal error occurred.',
          }),
        );
      }),
    );
    const { getByTestId, findByLabelText } = renderWithRedux(
      <CheckInScreen navigation={{ navigate, goBack }} />,
      {
        initialState: {
          User: {
            current_questionnaire_id:
              'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel|1.0',
            status: 'on-study',
            start_date: new Date(),
            firstTime: true,
            due_date: new Date(new Date().setDate(new Date().getDate() + 3)),
            certificate: appConfig.defaultRecipientCertificatePemString,
            additional_iterations_left: 0,
          },
          Globals: { loading: false },
        },
      },
    );
    findByLabelText(en.accessibility.refresh).then((refreshBtn) => {
      fireEvent.press(refreshBtn);
      waitForElementToBeRemoved(() => getByTestId('Spinner')).then(() => {
        expect(spyAlert).toHaveBeenCalledWith(
          en.generic.errorTitle,
          en.generic.updateError,
          expect.anything(),
          expect.anything(),
        );
        done();
      });
    });
  });

  it('should alert when submission of questionnaire failed', (done) => {
    const navigate = jest.fn();
    const goBack = jest.fn();
    const spyAlert = jest
      .spyOn(Alert, 'alert')
      .mockImplementation((_title, _message, callbackOrButtons) => {
        if (callbackOrButtons && callbackOrButtons[0].onPress) {
          callbackOrButtons[0].onPress();
        }
      });
    server.use(
      rest.post(`${endpoints.sendQuestionnaire}`, (_req, res, ctx) => {
        return res(
          ctx.status(404),
          ctx.json({
            errorCode: 'InternalErr',
            errorMessage: 'An internal error occurred.',
          }),
        );
      }),
    );
    const { findByText, getByTestId } = renderWithRedux(
      <CheckInScreen navigation={{ navigate, goBack }} />,
      {
        initialState: {
          User: {
            subjectId: 'someUserId',
            current_questionnaire_id:
              'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel|1.0',
            status: 'on-study',
            start_date: new Date(),
            firstTime: true,
            due_date: new Date(new Date().setDate(new Date().getDate() + 3)),
            certificate: appConfig.defaultRecipientCertificatePemString,
            additional_iterations_left: 1,
          },
          Questionnaire: {
            itemMap,
            categories,
            FHIRmetadata: {
              url: 'http://hl7.org/fhir/Questionnaire/Fragebogen_COMPASS_Beispiel',
              version: '1.0',
            },
          },
        },
      },
    );
    findByText(en.survey.send).then((btn) => {
      fireEvent.press(btn);
      waitForElementToBeRemoved(() => getByTestId('Spinner')).then(() => {
        expect(spyAlert).toHaveBeenCalledWith(
          en.generic.errorTitle,
          en.generic.sendError,
          expect.anything(),
          expect.anything(),
        );
        done();
      });
    });
  });

  it('should connect to FCM & send token to backend', (done) => {
    appConfig.connectToFCM = true;
    const navigate = jest.fn();
    const goBack = jest.fn();
    const spyRest = jest.spyOn(loggedInClient, 'updateDeviceToken');
    renderWithRedux(<CheckInScreen navigation={{ navigate, goBack }} />, {
      initialState: {
        User: { subjectId: 'someUserId', additional_iterations_left: 0 },
      },
    });

    waitFor(() => expect(spyRest).toHaveBeenCalled()).then(() => {
      done();
    });
  });
});
