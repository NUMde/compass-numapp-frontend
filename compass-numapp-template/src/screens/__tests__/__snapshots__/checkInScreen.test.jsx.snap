// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckInScreen should render the CheckInScreen 1`] = `
<View
  style={
    {
      "backgroundColor": "#F8F9FA",
      "flexDirection": "column",
      "height": "100%",
    }
  }
  testID="CheckInScreen"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "backgroundColor": "#F8F9FA",
        "flexDirection": "column",
        "height": 354.06976744186045,
        "paddingTop": 30,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "process": [Function],
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": 354.06976744186045,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
          "width": "90%",
        }
      }
    >
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityHint="Re-fetches the questionnaire and status. The current page will be updated."
          accessibilityLabel="refresh"
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="banner_refresh_btn"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#EB576A",
                    "fontSize": 48.83720930232558,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="iconIcon"
            >
              󰑐
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <Text
          accessibilityRole="header"
          numberOfLines={1}
          style={
            {
              "color": "#1F2532",
              "fontFamily": "IBMPlexSans-SemiBold",
              "fontSize": 33.75,
              "lineHeight": 43.125,
            }
          }
        >
          Check-In
        </Text>
      </View>
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityHint="Opens new page with additional information and links."
          accessibilityLabel="Additional Information"
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="banner_menu_btn"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#EB576A",
                    "fontSize": 48.83720930232558,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="iconIcon"
            >
              󰍜
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          {
            "process": [Function],
          }
        }
        style={
          {
            "alignSelf": "center",
            "bottom": 13.081395348837209,
            "maxHeight": 308.781773931855,
            "maxWidth": 650,
            "position": "relative",
          }
        }
      />
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
      enableOnAndroid={true}
      onScroll={[Function]}
      scrollEventThrottle={16}
      style={
        {
          "flex": 1,
          "flexDirection": "column",
          "width": "100%",
        }
      }
    >
      <View>
        <View
          style={
            [
              {
                "backgroundColor": "#F8F9FA",
                "flexDirection": "column",
                "height": "100%",
              },
              {
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <View
            accessibilityHint="Opens new page to see the questionnaire's categories.This questionnaire has not been answered yet."
            accessibilityLabel="Your Entry-Questionnaire Title. To be completed till:  03/14/2022"
            accessibilityRole="button"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="CheckInListItem"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#EB576A",
                  "borderBottomColor": "#EB576A",
                  "borderColor": "#bcbbc1",
                  "borderTopColor": "#EB576A",
                  "flexDirection": "row",
                  "padding": 52.325581395348834,
                  "width": "100%",
                }
              }
              testID="padView"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "flex-start",
                      "flex": 1,
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                theme={
                  {
                    "colors": {
                      "black": "#242424",
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": {
                        "android": {
                          "error": "#f44336",
                          "grey": "rgba(0, 0, 0, 0.54)",
                          "primary": "#2196f3",
                          "searchBg": "#dcdce1",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "default": {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "ios": {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "web": {
                          "error": "#ff190c",
                          "grey": "#393e42",
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                      "white": "#ffffff",
                    },
                  }
                }
              >
                <Text
                  style={
                    {
                      "backgroundColor": "transparent",
                      "color": "#F8F9FA",
                      "fontFamily": "IBMPlexSans-SemiBold",
                      "fontSize": 30,
                      "lineHeight": 37.5,
                    }
                  }
                  testID="listItemTitle"
                >
                  Your Entry-Questionnaire Title
                </Text>
                <Text
                  style={
                    {
                      "backgroundColor": "transparent",
                      "color": "#F8F9FA",
                      "fontFamily": "IBMPlexSans",
                      "fontSize": 28.125,
                      "lineHeight": 37.5,
                    }
                  }
                  testID="listItemTitle"
                >
                  To be completed till:  03/14/2022
                </Text>
              </View>
              <View
                style={
                  {
                    "paddingLeft": 16,
                  }
                }
              />
              <View
                style={
                  {
                    "alignSelf": "center",
                    "backgroundColor": "#F8F9FA",
                    "borderRadius": 16,
                    "height": 28,
                    "margin": 7,
                    "overflow": "hidden",
                    "shadowColor": "rgba(0,0,0, .4)",
                    "shadowOffset": {
                      "height": 1,
                      "width": 1,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 1,
                    "width": 28,
                  }
                }
              >
                <View>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#ffffff",
                        "borderRadius": 16,
                        "height": 28,
                        "justifyContent": "center",
                        "width": 28,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#EB576A",
                            "fontSize": 12,
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      󰲶
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "marginHorizontal": 52.325581395348834,
                "marginVertical": 43.604651162790695,
              }
            }
          >
            <View>
              <Text
                style={
                  {
                    "alignSelf": "center",
                    "color": "#1F2532",
                    "fontFamily": "IBMPlexSans-SemiBold",
                    "fontSize": 45,
                    "lineHeight": 54.375,
                    "textAlign": "center",
                  }
                }
              >
                CheckIn-Title for new Users
              </Text>
              <Text
                style={
                  {
                    "alignSelf": "center",
                    "color": "#1F2532",
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 28.125,
                    "lineHeight": 37.5,
                    "marginVertical": 17.441860465116278,
                    "textAlign": "center",
                  }
                }
              >
                This is text that is presented to first-time-users. It tells them when the first questionnaire is due: 
                <Text
                  style={
                    {
                      "fontFamily": "IBMPlexSans-SemiBold",
                      "fontSize": 26.25,
                      "lineHeight": 35.625,
                    }
                  }
                >
                  03/14/2022, 00:00
                  .
                </Text>
                 - See? This is the second part of that message.
              </Text>
              <Text
                style={
                  {
                    "alignSelf": "center",
                    "color": "#1F2532",
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 28.125,
                    "lineHeight": 37.5,
                    "marginVertical": 17.441860465116278,
                    "textAlign": "center",
                  }
                }
              >
                This is the bottom snippet of the welcomeText. You may place something useful here... like a limerick.
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#DADADA",
                  "borderRadius": 5,
                  "color": "white",
                  "display": "flex",
                  "height": 191.86046511627907,
                  "justifyContent": "center",
                  "margin": 23.25581395348837,
                  "opacity": 1,
                  "padding": 10,
                  "width": 305.2325581395349,
                }
              }
              testID="send_report_btn"
            >
              <View
                style={
                  {
                    "justifyContent": "space-around",
                  }
                }
              >
                <View
                  style={
                    {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          [
                            {
                              "color": "#F8F9FA",
                              "fontSize": 24,
                            },
                            {
                              "backgroundColor": "transparent",
                              "marginBottom": 5,
                            },
                            {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
                <Text
                  style={
                    {
                      "color": "white",
                      "fontFamily": "IBMPlexSans-SemiBold",
                      "fontSize": 26.25,
                      "lineHeight": 35.625,
                      "textAlign": "center",
                    }
                  }
                >
                  Button: Send out Special Report
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
